src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2742,6 +2742,36 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
             $container->removeDefinition('mailer.messenger_transport_listener');
         }
 
+        if ($config['dkim_signer']['enabled']) {
+            $dkimSigner = $container->getDefinition('mailer.dkim_signer');
+            $dkimSigner->setArgument(0, $config['dkim_signer']['key'] ?? null);
+            $dkimSigner->setArgument(1, $config['dkim_signer']['domain'] ?? null);
+            $dkimSigner->setArgument(2, $config['dkim_signer']['select'] ?? null);
```

Comment:
`null` is invalid for all all settings. The Configuration class needs to have validation rules enforcing those settings to be configured when the feature is enabled.